backend/services/mgmt/v1alpha1/transformers-service/transformers.go

```diff
@@ -723,3 +725,33 @@ func (s *Service) IsTransformerNameAvailable(ctx context.Context, req *connect.R
 	}), nil
 
 }
+
+// use the goja library to validate that the javascript can compile and theoretically run
+func (s *Service) ValidateUserJavascriptCode(ctx context.Context, req *connect.Request[mgmtv1alpha1.ValidateUserJavascriptCodeRequest]) (*connect.Response[mgmtv1alpha1.ValidateUserJavascriptCodeResponse], error) {
+
+	_, err := s.verifyUserInAccount(ctx, req.Msg.AccountId)
+	if err != nil {
+		return nil, err
+	}
+
+	js := constructJavascriptCode(req.Msg.Code, "test")
+
+	_, err = goja.Compile("test", js, true)
+	if err != nil {
+		return connect.NewResponse(&mgmtv1alpha1.ValidateUserJavascriptCodeResponse{
+			Valid: false,
+		}), nil
+	}
+
+	return connect.NewResponse(&mgmtv1alpha1.ValidateUserJavascriptCodeResponse{
+		Valid: true,
+	}), nil
+}
+
+func constructJavascriptCode(jsCode, col string) string {
+	if jsCode != "" {
+		return fmt.Sprintf(`(()=>{function fn1(value){%s};const input = benthos.v0_msg_as_structured();const output = { ...input };output["%[2]s"] = fn1(input["%[2]s"]);benthos.v0_msg_set_structured(output);})();`, jsCode, col)
```

Comment:
Can we simplify this function and move it to multiple lines so it's easier to read? It's a multiline string anyways.
```
return fmt.Sprintf(`(() => {
    function fn1(value) {
      %s
    }
  })
`, userCode)
```

Does it need to be more than that for a simple test?

Reply:
was trying to be consistent but simplified it